engines/ags/shared/gfx/allegro_bitmap.cpp

```diff
@@ -128,8 +128,27 @@ bool Bitmap::LoadFromFile(const char *filename) {
 
 	BITMAP *al_bmp = load_bitmap(filename, nullptr);
 	if (al_bmp) {
-		_alBitmap = al_bmp;
-		_isDataOwner = true;
+		Graphics::ManagedSurface *origSurface = new Graphics::ManagedSurface(al_bmp->getSurface());
+		Graphics::ManagedSurface *newSurface = nullptr;
+
+		if (origSurface->format.bytesPerPixel == 2 && origSurface->format != Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0)) { // convert to RGB565
+			newSurface = new Graphics::ManagedSurface(origSurface->w, origSurface->h, Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0));
+		} else if ((origSurface->format.bytesPerPixel == 3) || (origSurface->format.bytesPerPixel == 4 && origSurface->format != Graphics::PixelFormat(4, 8, 8, 8, 8, 16, 8, 0, 24))) { // convert to RGBA8888
+			newSurface = new Graphics::ManagedSurface(origSurface->w, origSurface->h, Graphics::PixelFormat(4, 8, 8, 8, 8, 16, 8, 0, 24));
+		} else { // no conversion needed
+			_alBitmap = al_bmp;
+			_isDataOwner = true;
+			return _alBitmap != nullptr;
+		}
+
+		if (newSurface) {
+			newSurface->blitFrom(*origSurface);
+			BITMAP *conv_al_bmp = new BITMAP(newSurface);
+			if (conv_al_bmp) {
+				_alBitmap = conv_al_bmp;
+				_isDataOwner = true;
+			}
+		}
```

Comment:
Since these two code blocks are exactly the same, it makes sense to create an internal reusable function, which accepts a an argument of type BITMAP * and contains all this new code

Reply:
Good idea,  I moved the conversion code to a new function